src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/PaymentsCallbackPBAHandler.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    private CallbackResponse makePbaPayment(CallbackParams callbackParams) {
+        var caseData = callbackParams.getCaseData();
+        var authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        List<String> errors = new ArrayList<>();
+        try {
+            log.info("processing payment for case " + caseData.getCcdCaseReference());
+            var paymentReference = paymentsService.createCreditAccountPayment1(caseData, authToken)
+                .getPaymentReference();
+
+            PaymentDetails paymentDetails = ofNullable(caseData.getClaimIssuedPaymentDetails())
+                .map(PaymentDetails::toBuilder)
+                .orElse(PaymentDetails.builder())
+                .status(SUCCESS)
+                .reference(paymentReference)
+                .errorCode(null)
+                .errorMessage(null)
+                .build();
+
+            caseData = caseData.toBuilder()
+                .claimIssuedPaymentDetails(paymentDetails)
+                .paymentSuccessfulDate(time.now())
+                .build();
+
+        } catch (FeignException e) {
+            log.info(String.format("Http Status %s ", e.status()), e);
+            if (e.status() == 403 || e.status() == 422 || e.status() == 504) {
+                caseData = updateWithBusinessError(caseData, e);
+            } else {
+                errors.add(ERROR_MESSAGE);
+            }
+        } catch (InvalidPaymentRequestException e) {
+            log.error(String.format("Duplicate Payment error status code 400 for case: %s, response body: %s",
+                                    caseData.getCcdCaseReference(), e.getMessage()
+            ));
+            caseData = updateWithDuplicatePaymentError(caseData);
+        }
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseData.toMap(objectMapper))
+            .errors(errors)
+            .build();
+    }
+
+    private CaseData updateWithBusinessError(CaseData caseData, FeignException e) {
+        try {
+            var paymentDto = objectMapper.readValue(e.contentUTF8(), PaymentDto.class);
+            var statusHistory = paymentDto.getStatusHistories()[0];
+            PaymentDetails paymentDetailsErrored = ofNullable(caseData.getClaimIssuedPaymentDetails())
+                .map(PaymentDetails::toBuilder).orElse(PaymentDetails.builder())
+                .status(FAILED)
+                .errorCode(statusHistory.getErrorCode())
+                .errorMessage(statusHistory.getErrorMessage())
+                .build();
+
+            return caseData.toBuilder()
+                .claimIssuedPaymentDetails(paymentDetailsErrored)
+                .build();
+        } catch (JsonProcessingException jsonException) {
+            log.error(jsonException.getMessage());
+            log.error(String.format("Unknown payment error for case: %s, response body: %s",
+                                    caseData.getCcdCaseReference(), e.contentUTF8()
```

Comment:
No need for `String.format`. See comments above.

Reply:
I removed this class, as it was added, but did not seem to be used anywhere. Resolved myself in your absence.